tests/o2a_libs/test_el_parser.py

```diff
@@ -0,0 +1,118 @@
+            ),
+            ("${YEAR}/${MONTH}/${DAY}/${HOUR}", "{{year}}/{{month}}/{{day}}/{{hour}}"),
+            ("pure text without any.function wf:function()", "pure text without any.function wf:function()"),
+            (
+                "${fs:fileSize('/usr/foo/myinputdir') gt 10 * GB}",
+                "{{fs_file_size('/usr/foo/myinputdir') > 10 * gb}}",
+            ),
+            (
+                '${hadoop:counters("mr-node")["FileSystemCounters"]["FILE_BYTES_READ"]}',
+                '{{hadoop_counters("mr-node")["FileSystemCounters"]["FILE_BYTES_READ"]}}',
+            ),
+            ('${hadoop:counters("pig-node")["JOB_GRAPH"]}', '{{hadoop_counters("pig-node")["JOB_GRAPH"]}}'),
+            (
+                "${wf:actionData('shell-node')['my_output'] eq 'Hello Oozie'}",
+                "{{wf_action_data('shell-node')['my_output'] == 'Hello Oozie'}}",
+            ),
+            ("${coord:dataOut('output')}", "{{coord_data_out('output')}}"),
+            ("${coord:current(0)}", "{{coord_current(0)}}"),
+            ('${coord:offset(-42, "MINUTE")}', '{{coord_offset(-42,"MINUTE")}}'),
+            (
+                "${(wf:actionData('java1')['datelist'] == EXPECTED_DATE_RANGE)}",
+                "{{(wf_action_data('java1')['datelist'] == expected_date_range)}}",
+            ),
+            (
+                '${(hadoop:counters("mr-node")[RECORDS][MAP_IN] == hadoop:counters("mr-node")'
+                '[RECORDS][MAP_OUT]) and (hadoop:counters("mr-node")[RECORDS][REDUCE_IN] == '
+                'hadoop:counters("mr-node")[RECORDS][REDUCE_OUT]) and (hadoop:counters("mr-node")'
+                "[RECORDS][GROUPS] gt 0)}",
+                '{{(hadoop_counters("mr-node")[records][map_in] == hadoop_counters("mr-node")'
+                '[records][map_out]) and (hadoop_counters("mr-node")[records][reduce_in] == '
+                'hadoop_counters("mr-node")[records][reduce_out]) and (hadoop_counters("mr-node")'
+                "[records][groups] > 0)}}",
+            ),
+            (
+                "${fs:exists(concat(concat(concat(concat(concat(nameNode, '/user/'),"
+                "wf:user()), '/'), examplesRoot), '/output-data/demo/mr-node')) == 'true'}",
+                "{{fs_exists(concat(concat(concat(concat(concat(name_node,'/user/'),"
+                "wf_user()),'/'),examples_root),'/output-data/demo/mr-node')) == 'true'}}",
+            ),
+            (
+                "${wf:actionData('getDirInfo')['dir.num-files'] gt 23 || "
+                "wf:actionData('getDirInfo')['dir.age'] gt 6}",
+                "{{wf_action_data('getDirInfo')['dir.num-files'] > 23 or "
+                "wf_action_data('getDirInfo')['dir.age'] > 6}}",
+            ),
+            ("#{'${'}", "{{'${'}}"),
+            ("${'${'}", "{{'${'}}"),
+            ("${function()} literal and #{OtherFunction}", "{{function()}} literal and{{other_function}}"),
```

Comment:
I am not sure if this example is parsed correctly.  Should we skip the space before Jinja expression?
`...and{{..." -> "...and {{...`